lib/spack/spack/ci.py

```diff
@@ -627,12 +666,31 @@ def generate_gitlab_ci_yaml(env, print_summary, output_file,
                         bs_arch = bs['spec'].architecture
                         if (bs['spec'].satisfies(compiler_pkg_spec) and
                             bs_arch == release_spec.architecture):
-                            c_job_name = get_job_name(bs['phase-name'],
-                                                      bs['strip-compilers'],
-                                                      bs['spec'],
-                                                      str(bs_arch),
-                                                      build_group)
-                            job_dependencies.append(c_job_name)
+                            # We found the bootstrap compiler this release spec
+                            # should be built with, so for DAG scheduling
+                            # purposes, we will at least add the compiler spec
+                            # to the jobs "needs", but if artifact buildcache
+                            # is enabled, we'll have to add all transtive deps
+                            # of the compiler as well.
+                            if enable_artifacts_buildcache:
```

Comment:
It looks like the `enable_artifacts_buildcache` branch is not tested (from the covrerage report).  Can you add a test for this part?

Reply:
Based on the coverage, it looks to me like the branch above is the one that's never tested ( this `if` and the following `else` are never hit).  After looking into it, the reason for that is my test bootstraps `cmake` instead of a compiler (probably because I didn't know how to do what I'm about to ask).

 In order to test these blocks, I think need to find a mock package that can be "built" with a mock compiler, so that eventually this code discovers that a bootstrapped compiler `satisfies` the compiler required to build the spec.  I tried bootstrapping the mock `gcc` and using it as the compiler for some other packages, but so far, that mock `gcc` still doesn't "satisfy" the compiler of the mock package.  Do you know if any mock packages can be tested as built with a mocked compiler?

Reply:
I added a couple tests to make sure we're generating the right `needs` list for compiler bootstrapping pipelines, when `enable-artifacts-buildcache` is either `True` or `False`.